docs/secret-rotation.md

```diff
@@ -0,0 +1,254 @@
+### **Create/Delete Credentials**
+
+The first step is to access the Horizon application credentials in order to either create a new one or to delete the old, compromised secret.
+
+Application credentials can be managed through the use of the openstack CLI which can be found [here](https://docs.openstack.org/keystone/2024.2/user/application_credentials.html), however, for the sake of simplicity, the following steps will outline how to do this through the Horizon dashboard user interface.
+
+Once logged into Horizon, head to the **Identity** tab on the left, then **Application Credentials**. From this page, it is possible to manage the secrets which authenticate operations targetted at the OpenStack system.
+
+| If the credential’s been compromised, this is the point in which the application credential will need to be deleted. |
+| :---- |
+
+In the case of creating a new set of application credentials, please remember to download the **clouds.yaml** once the configuration steps are complete; this will be the only time you will be able to do so, otherwise a new set of credentials will need to be created.
+
+At this point, the newly created secret requires encrypting before it can be rotated in.
+
+<a id="processing-secrets-link"></a>
+# **Processing Secrets**
+
+<a id="encrypting-new-credential-link"></a>
+### **Encrypting New Credential**
+
+Once the new secret **clouds.yaml** has been downloaded, and before going any further, it would be a good idea to rename the old **credentials.yaml** in `fluxcd-config/clusters/<cluster-name>/` into something like **old-credentials.yaml** \- *of course, only if the secret hasn’t already been deleted in Horizon or is suspected to be compromised*.
+
+Now a new file called **credentials.yaml** can be created in its place with the following structure:
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <MAKE SURE THIS MATCHES “cloudCredentialsSecretName:” IN configmap.yaml>
+  namespace: capi-self
+  annotations:
+    # Allow the sealed secret controller to take over this secret after bootstrapping
+    sealedsecrets.bitnami.com/managed: "true"
+stringData:
+  clouds.yaml: |
+    <INSERT THE CONTENTS OF YOUR NEW SECRET’S CLOUDS.YAML, PAYING ATTENTION TO THE
+    INDENTATIONS>
+```
+
+Once the newly created **clouds.yaml**’s contents have been copied into **credentials.yaml**, it should look something like this:
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <MAKE SURE THIS MATCHES “cloudCredentialsSecretName:” IN configmap.yaml>
+  namespace: capi-self
+  annotations:
+    # Allow the sealed secret controller to take over this secret after bootstrapping
+    sealedsecrets.bitnami.com/managed: "true"
+stringData:
+  clouds.yaml: |
+    clouds:
+      openstack:
+        auth:
+          auth_url: ...
+          application_credential_id: ...
+          application_credential_secret: ...
+        interface: ...
+        identity_api_version: 3
+        auth_type: ...
```

Comment:
This is missing the `project_id` field as mentioned [here](https://github.com/stackhpc/capi-helm-fluxcd-config/tree/secret-rotation-docs?tab=readme-ov-file#usage) and explained in more detail [here](https://github.com/azimuth-cloud/capi-helm-charts/blob/main/charts/openstack-cluster/README.md#openstack-credentials). Not sure if / how you managed to test this secret rotation process without the `project_id` field in your new credentials? Unless the project ID requirement has been removed in the upstream `gophercloud` library and I've just never noticed 😄

Reply:
We discussed this previously